### DISCUSSION TITLE
Multiple homespan devices

### DISCUSSION THREAD

**Discussion by florianHK** [HomeSpan/HomeSpan#197] [Q&A]
Is there anything to change in the files if you have multiple esp32 homespan devices, e.g. the Homekit code or Serial number? After installing a second ESP device the first one was not responding in Homekit.

BTW - Thanks a lot for the great code! This is an absolute awesome project! I finally have a 5 RGB LED status monitor to monitor my locks at home.

**Comment by HomeSpan** [comment]
If you don't change any of the default settings, HomeSpan will create random HomeKit device IDs as well as use those device IDs as the suffix for the hostname of the device, which basically guarantees uniqueness.  In principle the HomeKit name of two devices can be the same, but I just tried and found I could not pair the second device if I used the same HomeKit name as defined in the homeSpan.begin() method.  To ensure the devices have a different HomeKit name, change the names in homeSpan.begin() as follows:

```C++
  homeSpan.begin(Category::Lighting,"HomeSpan LightBulb 1");   // USE THIS FOR ONE DEVICE
  homeSpan.begin(Category::Lighting,"HomeSpan LightBulb 2");   // USE THIS FOR A SECOND DEVICE
 ```

To test I loaded the Example 1 LightBulb tutorial onto two different ESP32s where the only difference is in the name as per above.  I was able to pair and operate both devices with only this change.

If you are having problems with a particular device, such as the Home App not being able to find it when trying to pair, it may be that the device was previously paired.  To reset all pairing data and create a fresh random device ID, type "H" into the CLI for a particular device.  This will not reset the WiFi Credentials, just the pairing information.

**Reply by florianHK** [reply]
Thanks a lot for the quick response, will try it tomorrow.

**Reply by royfocker19** [reply]
Is it possible to autogenerate the serial number based on the MAC address of the device?

**Comment by HomeSpan** [comment]
Not at present - the HAP Device ID is randomly-generated and separate from the MAC address.  Are you looking to be able to specify the HAP Device ID (based on MAC or any other bytes you choose)?

**Comment by royfocker19** [comment]
I want to be able to see the MAC address somewhere in the information of the device. I think the serial number is the best place to have it. I have a function working based on the wifi sdk. I’ll try it and let you know if it works 

Sent from my iPhone 

Press in Dire Situations http://www.nooooooooooooooo.com/
